src/AvatarStack.tsx

```diff
@@ -140,17 +137,18 @@ const transformChildren = (children: React.ReactNode) => {
 
 export type AvatarStackProps = {
   alignRight?: boolean
-} & ComponentProps<typeof AvatarStackWrapper>
+  children: React.ReactNode
+} & SxProp
 
-const AvatarStack = ({children, alignRight, ...rest}: AvatarStackProps) => {
+const AvatarStack = ({children, alignRight, sx: sxProp}: AvatarStackProps) => {
   const count = React.Children.count(children)
   const wrapperClassNames = classnames({
     'pc-AvatarStack--two': count === 2,
     'pc-AvatarStack--three-plus': count > 2,
     'pc-AvatarStack--right': alignRight
   })
   return (
-    <AvatarStackWrapper count={count} className={wrapperClassNames} {...rest}>
```

Comment:
This change prevents AvatarStack from accepting any DOM props (e.g. `id`, `className`, `data-*`). I would love for us to document how we decide when a component should accept DOM props. Maybe it's appropriate for some components to accept all DOM props (e.g. `Link`) and other components should only accept a subset of DOM props. 

We should also make it obvious in our component documentation pages if a component accepts DOM props or not. 

This is a bigger discussion that we can probably have outside of this PR.

For now, I'm fine with moving forward with this PR as-is. It might help us expose places where DOM props are necessary.

Reply:
Good call to make it clear in docs.

Proposal: The set of components that take an `as` prop should be the same set that pass through DOM props. (I can't see any use case where we'd want those sets to diverge).

Reply:
That seems reasonable to me. I'm trying to think of components that might not accept the `as` prop but should still accept DOM props. What about form components? (like `TextInput`)